projects/plugins/jetpack/tests/php/_inc/lib/test_class.rest-api-endpoints.php

```diff
@@ -1126,4 +1126,103 @@ public function test_get_user_connection_data_without_master_user() {
 		$response_data = $response->get_data();
 		$this->assertNull( $response_data['connectionOwner'] );
 	}
+
+	/**
+	 * Test fetching a site's purchase token.
+	 *
+	 * @since 9.8.0
+	 */
+	public function test_get_purchase_token() {
```

Comment:
Could we add two more cases here pls?
- Same endpoint, blog token -> should succeed
- Same endpoint, user without permissions -> should fail

Reply:
To facilitate development because of code freeze I added the blog token related tests in https://github.com/Automattic/jetpack/pull/19924/commits/541a219807ff8e356a23976d2f651a399e5e803d

Note that I added those tests in `WP_Test_Jetpack_REST_API_Authentication` because in this case we are interested in the more generic concept of auth via blog token, not what the endpoint actually returns.

You can add your tests (user without permissions) in `test_class.rest-api-endpoints.php` along with your existing ones!

Reply:
I added tests to cover the `Same endpoint, user without permissions -> should fail` case on both endpoints, but I failed at covering the `Same endpoint, blog token -> should succeed` case. I think I'm not sending the blog token as I should and because of that, the endpoint returns a 401 rather than a 200. I would appreciate very much your guidance on this issue. This is the failed test I added: https://github.com/Automattic/jetpack/pull/19924/files#diff-8682eefbe80771f466c87162b6c1a1c9e7768daa94014285e575a7a30c3f1193R178.

Reply:
No worries, the `WP_Test_Jetpack_REST_API_Authentication` should be enough.